### ISSUE TITLE
Parse nested lists

### ISSUE THREAD

**Issue by dillonkearns** [dillonkearns/elm-markdown#7]
Currently the [List Block type](https://package.elm-lang.org/packages/dillonkearns/elm-markdown/latest/Markdown-Block) just has inlines:

```
type Block =
   -- ... other blocks
    | ListBlock (List (List Inline))
```

The `Block` type will need to change to support this. It should support having nested `Block`s, including other lists.

An example of a failing test cases for this is https://github.com/dillonkearns/elm-markdown/blob/master/test-results/failing/GFM/Lists.md#example-277.

**Comment by heyakyra** [comment]
No workaround for this?

**Comment by klaftertief** [comment]
I started working on this a bit, but had to shift focus. Will try to come back to it, as I need it soonish.

**Comment by dillonkearns** [comment]
This was fixed by #88! 🎉 Thanks again @LutSa and @klaftertief!!!
